tm2/pkg/telemetry/metrics/metrics.go

```diff
@@ -124,6 +128,43 @@ func Init(config config.Config) error {
 		return fmt.Errorf("error parsing exporter endpoint: %s, %w", config.ExporterEndpoint, err)
 	}
 
+	// The exporter embeds a default OpenTelemetry Reader and
+	// implements prometheus.Collector, allowing it to be used as
+	// both a Reader and Collector.
+	promexp, err := prometheus.New()
+	if err != nil {
+		return fmt.Errorf("error creating prometheus exporter, %w", err)
+	}
+
+	providerOptions := []sdkMetric.Option{
+		sdkMetric.WithResource(
+			resource.NewWithAttributes(
+				semconv.SchemaURL,
+				semconv.ServiceNameKey.String(config.ServiceName),
+				// TODO: Get git tag / commit version
+				semconv.ServiceVersionKey.String("1.0.0"),
+				semconv.ServiceInstanceIDKey.String(config.ServiceInstance),
+			),
+		),
+	}
+
+	if config.PrometheusAddr != "" {
+		providerOptions = append(providerOptions, sdkMetric.WithReader(promexp))
+
+		go func() {
+			server := &http.Server{
+				Addr:              config.PrometheusAddr,
+				ReadHeaderTimeout: 5 * time.Second,
+			}
+			http.Handle("/metrics", promhttp.Handler())
+
+			err := server.ListenAndServe()
+			if err != nil {
+				fmt.Printf("[ERROR] Prometheus metrics server error: %v\n", err)
+			}
+		}()
```

Comment:
There needs to be better management of this go routine :(

What do you think about tying it to the context that initializes the metrics themselves?